website_form_validations/controllers/main.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+from openerp import SUPERUSER_ID
+from openerp.http import request
+from openerp.addons.website_sale.controllers.main import website_sale
+
+
+class website_sale_inh(website_sale):
+
+    def checkout_values(self, data=None):
+        cr, uid, context, registry =\
+            request.cr, request.uid, request.context, request.registry
+        orm_partner = registry.get('res.partner')
+        orm_user = registry.get('res.users')
+        orm_country = registry.get('res.country')
+        state_orm = registry.get('res.country.state')
+
+        country_ids = orm_country.search(cr, SUPERUSER_ID, [], context=context)
+        countries = orm_country.browse(cr, SUPERUSER_ID, country_ids, context)
+        states_ids = state_orm.search(cr, SUPERUSER_ID, [], context=context)
+        states = state_orm.browse(cr, SUPERUSER_ID, states_ids, context)
+        partner = orm_user.browse(
+            cr,
+            SUPERUSER_ID,
+            request.uid,
+            context).partner_id
```

Comment:
Wrong, set in the controller a browseable with superuser id is a security fault, can you sel a logic in object side for such deal.